### DISCUSSION TITLE
Do not disturb/global notification supression

### DISCUSSION THREAD

**Discussion by jupjohn** [Chatterino/chatterino2#5628] [Ideas]
![image](https://github.com/user-attachments/assets/ce2aaad6-7a7b-4e93-84b8-a960fbd8b5cf)

Possible current solutions I can think of:
 - Mute the application in your OSes volume mixer (windows volume mixer, pavucontrol etc.)
 - Manually enable streamer mode when wanting to mute notifications
 - Just close chatterino

**Comment by pajlada** [comment]
When enabling DND, would you want the DND setting to stay after a restart of Chatterino?

**Comment by pajlada** [comment]
What would you like to prevent Chatterino from doing when DND is enabled?

**Comment by pajlada** [comment]
Do you consider DND a global feature, like setting yourself do Busy in Discord, or a more local feature, like muting a server in Discord?

**Comment by brian6932** [comment]
> When enabling DND, would you want the DND setting to stay after a restart of Chatterino?

Probably, but I think this value would have some priority in saving to the json file. The hypothetical would be you turned off DND, the client crashes, or somehow gets killed, DND could randomly turn itself back on, and that would be problematic.

> What would you like to prevent Chatterino from doing when DND is enabled?

> Do you consider DND a global feature, like setting yourself do Busy in Discord, or a more local feature, like muting a server in Discord?

I don't think this should toggle on streamer mode, but it should suppress all notifications (toasts from streams going live, taskbar flashes, ping sounds). Instead, the way I think this should work would be to collect `/mentions` & `/whispers` with a label while in DND, kind of like the `Draw a line below the most recent message before switching applications.` feature, or some badge/text, with a tag that could be filtered by <kbd>CTRL</kbd>+<kbd>F</kbd>->`is:dnd`. Additionally, it should really have a global bind (e.g. Pressing <kbd>F24</kbd> no matter the window focus toggles DND).
